src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -4148,23 +4148,148 @@ private Position endPosition()
       return Position.create(endRow, endColumn);
    }
    
+   // splits a chunk into two chunks
+   // chunk 1: first chunk line to linePos (not including linePos)
+   // chunk 2: linePos line to end
+   private void splitChunk(Scope chunk, int linePos)
+   {
+      Position chunkStart = chunk.getBodyStart();
+      Position chunkEnd = chunk.getEnd();
+      Position preamble = chunk.getPreamble();
+      String preambleLine = docDisplay_.getTextForRange(Range.create(preamble.getRow(), preamble.getColumn(), preamble.getRow() + 1, 0));
+      
+      // if the cursor line is in the preamble of the chunk
+      // reset it to the body of the chunk to get the same semantics
+      // (empty chunk followed by the entire existing chunk)
+      if (linePos < chunkStart.getRow())
+         linePos = chunkStart.getRow();
+         
+      // get chunk contents from chunk start up to the specified line
+      Range firstChunkRange = Range.create(chunkStart.getRow(), chunkStart.getColumn(), linePos, 0);
+      String firstChunkContents = docDisplay_.getTextForRange(firstChunkRange);
+      firstChunkContents = firstChunkContents.trim();
+      
+      // add preamble line and ending line for new first chunk
+      firstChunkContents = preambleLine + firstChunkContents;
+      if (!firstChunkContents.endsWith("\n"))
+         firstChunkContents += "\n";  
+      firstChunkContents += "```\n\n";
```

Comment:
Should this function accept the chunk start + end delimiters as arguments? We use different chunk delimiters for different kinds of multimode documents, e.g.

#### R Markdown

    ```{options}
    ```

#### Sweave

    <<options>>=
    @ 

#### R HTML

    <!-- begin.rcode
    end.rcode -->

Unfortunately the starts and ends are currently stuffed together in most places where we have the 'skeleton' for a chunk; e.g.

https://github.com/rstudio/rstudio/blob/30ea712f4b02f84cc8269438cd8bcec8692d12ed/src/gwt/acesupport/acemode/rmarkdown.js#L120-L124